### PR TITLE
[WIP/For discussion]: Including parameters in python based launch file

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Right now to launch a MoveIt 2 node we will list all the parameters manually [example](https://github.com/ros-planning/moveit2/blob/master/moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py), this's a proposal to do something similar to ROS [example](https://github.com/ros-planning/moveit_resources/blob/master/panda_moveit_config/launch/move_group.launch)

This's based on this [commit](https://github.com/JafarAbdi/moveit_resources/commit/a1dc9b41df641c22fb35e258ddf1faf39b8a41de) and this [package](https://github.com/JafarAbdi/moveit_helpers)

I did try to declare the args in the other launch files using `DeclareLaunchArgument` and include them with `PythonLaunchDescriptionSource`/`IncludeLaunchDescription` and propagate the arguments with `get_launch_arguments()` but the solution was complicated and even it didn't work with arguments with non-default values 

I Opened a PR rather than an issue just so you can easily see the diff for this package

The naming and the structure of moveit_helpers need some changes, this's just a draft

### CODE REVIEW

**Comment by henningkayser** [comment]
@JafarAbdi thanks for starting this! It really looks much cleaner and I like the approach in general. My suggestion would be to use a specific name and structure convention for pure parameter files,  meaning the old `xml.launch` files should not be confused with `launch.py` files that actually construct a `LaunchDescription` and run the nodes. I'm thinking something like exposing a function `load_parameters` that you can than map into your desired namespace or node.

**Review by DLu — CHANGES_REQUESTED** [review]
1. We should open a ticket upstream to get the MoveIt! helpers merged into ROS 2 Launch. 
 2. This currently has conflicts, but otherwise looks good.

**Comment by DLu** [comment]
Also, are there plans to release `moveit_helpers`?

File: moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py [hunk 1]
```diff
@@ -3,51 +3,16 @@
...
-    package_path = get_package_share_directory(package_name)
-    absolute_file_path = os.path.join(package_path, file_path)
-
-    try:
-        with open(absolute_file_path, 'r') as file:
-            return yaml.load(file)
-    except EnvironmentError: # parent of IOError, OSError *and* WindowsError where available
-        return None
-
+from moveit_helpers.get_parameters_module import get_parameters_module, load_file, load_yaml
```
Comment (davetcoleman) [inline moveit_demo_nodes/run_moveit_cpp/launch/run_moveit_cpp.launch.py:6]: @DLu is wondering if this should be a standard helping function in ros2 launch, vs a moveit_helper function

**Review by davetcoleman — COMMENTED** [review]
This seems like a no-brainer code cleanup PR, just needs conflicts resolved.

**Comment by henningkayser** [comment]
> Also, are there plans to release `moveit_helpers`?

Not sure if this really makes a lot of sense at this point. While this PR includes some nice cleanup attempts, I'm not convinced that this is a good long-term solution. It doesn't help a lot with reducing launch file redundancy and we should probably rather use the native way of loading yaml files instead of `load_yaml()`.

**Comment by JafarAbdi** [comment]
Replaced by https://github.com/ros-planning/moveit2/pull/591

**Pull request closed without merging**